src/_gettsim_tests/test_aggregate_by_p_id.py

```diff
@@ -1,36 +1,33 @@
+import dags.tree as dt
 import pytest
-from pandas.testing import assert_series_equal
+from numpy.testing import assert_array_almost_equal
 
 from _gettsim.interface import compute_taxes_and_transfers
 from _gettsim_tests._helpers import cached_set_up_policy_environment
-from _gettsim_tests._policy_test_utils import PolicyTestData, load_policy_test_data
+from _gettsim_tests._policy_test_utils import PolicyTest, load_policy_test_data
 
-OVERRIDE_COLS = []
+test_data = load_policy_test_data("aggregate_by_p_id")
 
-data = load_policy_test_data("aggregate_by_p_id")
 
-
-@pytest.mark.xfail(reason="Needs renamings PR.")
 @pytest.mark.parametrize(
-    ("test_data", "column"),
-    data.parametrize_args,
-    ids=str,
+    "test",
+    test_data,
 )
 def test_aggregate_by_p_id(
-    test_data: PolicyTestData,
-    column: str,
+    test: PolicyTest,
 ):
-    df = test_data.input_df
-    environment = cached_set_up_policy_environment(date=test_data.date)
+    environment = cached_set_up_policy_environment(date=test.date)
 
     result = compute_taxes_and_transfers(
-        data=df, environment=environment, targets=column
+        data_tree=test.input_tree,
+        environment=environment,
+        targets_tree=test.target_structure,
     )
 
-    assert_series_equal(
-        result[column],
-        test_data.output_df[column],
-        check_dtype=False,
-        atol=1e-1,
-        rtol=0,
-    )
+    flat_result = dt.flatten_to_qual_names(result)
+    flat_expected_output_tree = dt.flatten_to_qual_names(test.expected_output_tree)
+
+    for result, expected in zip(
```

Comment:
Add to the prior issue that we move to comparing the trees directly rather than array-by-array. Should be pretty straightforward using `tree_map_with_path`, I'd hope...